src/SharpLearning.Optimization/GlobalizedBoundedNelderMeadOptimizer.cs

```diff
@@ -53,9 +55,10 @@ public sealed class GlobalizedBoundedNelderMeadOptimizer : IOptimizer
         /// <param name="rho">Coefficient for contraction part of the algorithm (default is -0.5)</param>
         /// <param name="sigma">Coefficient for shrink part of the algorithm (default is 0.5)</param>
         /// <param name="seed">Seed for random restarts</param>
+        /// <param name="maxDegreeOfParallelism">Maximum number of concurrent operations (default is unlimited)</param>
```

Comment:
@jameschch Nit picking: Consider specifying that the value for no bounds on the `maxDegreeOfParallelism` is -1.

Reply:
@jameschch Overall the pull request looks good, thank you very much for contribution! 

I imagine that you are not working on this in the weekend, so I will complete the PR in its current state, and fix the few small things I have commented on in the review, in a separate pull request. I have a few other smaller things I need to add to the optimization library anyway.

Thanks again,
Best regards 
Mads

Reply:
Yes, giving special significance to -1 is no longer a common thing.